### PR TITLE
Phase 2 GMT Tau->3Mu Emulator

### PR DESCRIPTION
**Pull request by zhenbinwu**
#### PR description:

This is the first version of the Phase2 GMT tau-> 3mu emulator and a minor update of the output object according to the updated interface. 

#### PR validation:

The emulators are matched to the firmware output.

### CODE REVIEW

File: L1Trigger/Phase2L1GMT/interface/Tauto3Mu.h [hunk 1]
```diff
@@ -84,68 +165,106 @@ namespace Phase2L1GMT {
...
+    float mass12 = GetDiMass(trkMus->at(target), trkMus->at(obj1));
+    float mass23 = GetDiMass(trkMus->at(obj1), trkMus->at(obj2));
+    float mass31 = GetDiMass(trkMus->at(obj2), trkMus->at(target));
 
-    return mass12 + mass23 + mass31;
+    ap_ufixed<45, 26, AP_TRN, AP_SAT> temp(mass12 + mass23 + mass31);
+    if (verbose_)
+    {
+      std::cout <<  target <<" "<<obj1 <<" "<<obj2 << std::endl;
+      std::cout << "mass12 " <<  mass12 <<", "
```
Comment (cecilecaillol) [inline L1Trigger/Phase2L1GMT/interface/Tauto3Mu.h:177]: Can you replace the couts by LogDebug/LogWarning/... as appropriate?

File: L1Trigger/Phase2L1GMT/interface/Tauto3Mu.h [hunk 2]
```diff
@@ -84,68 +165,106 @@ namespace Phase2L1GMT {
...
+    if (abovepihalf)
+        dphi = dphi - pihalf;
+    float cosdphi = cos_dphi_lut[dphi].to_float();
+    if (abovepihalf)
+        cosdphi = -1 * cosdphi;
+
+    float mass = ptprod * ( coshdeta - cosdphi);
+    if (verbose_)
+    {
+         std::cout << " ..... "
```
Comment (cecilecaillol) [inline L1Trigger/Phase2L1GMT/interface/Tauto3Mu.h:208]: Same, can you please use LogDebug/LogWarning/...?

**Comment by gpetruc** [comment]
Hi.
I was wondering if you plan to update this, either in 11_1_X or in 12_X: as this changes the standalone muon encoding, when this gets integrated we have to simultaneously change the standalone muon decoding in the correlator emulator and firmware

**Comment by cbotta** [comment]
Hi. It was not in the plan, given it doesn't yet provide realistic performance. But given what you just mention, it can be discussed in software meeting this week

**Comment by zhenbinwu** [comment]
Hi @gpetruc. I am still working on Tau->3mu part. But for small fix of the standalone muon encoding, I can make a new PR to the 12_X branch.

**Comment by zhenbinwu** [comment]
Ok. The new PR #995  has been created for  12_3_0_pre4 branch.

**Comment by epalencia** [comment]
@zhenbinwu, is this PR still needed? If not, could you please close it?

**Comment by zhenbinwu** [comment]
Closing this PR. I will create another PR from 12X

**Pull request closed without merging**